### PR TITLE
[Task] Added Apache SeaTunnel 2.x task node

### LINKED ISSUE #10422: [Feature][Task] Added Apache SeaTunnel 2.x task node
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Added Apache SeaTunnel 2.x task node

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of the pull request

this pr closes #10422 

1.Flink use resource config:
iShot2022-06-16_18 52 16

2.Flink use custom config:
iShot2022-06-16_18 52 50

3.Spark use custom config:
iShot2022-06-16_18 53 10

4.Spark use resource config:
iShot2022-06-16_18 53 35

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-ui/src/views/projects/task/components/node/types.ts [hunk 1]
```diff
@@ -396,6 +398,8 @@ interface INodeData
   udfs?: string[]
   customConfig?: boolean
   mapping_columns?: object[]
+  useCustom?: boolean
+  runMode?: string
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/projects/task/components/node/types.ts]: No need to define 'useCustom' and 'runMode' here if you have already defined them in ITaskParams as the same types.

Comment (zhuangchong) [reply dolphinscheduler-ui/src/views/projects/task/components/node/types.ts]: done. thx.

**Review by Amy0104 — DISMISSED** [review]
The front end part LGTM.

**Review by SbloodyS — DISMISSED** [review]
+1

File: script/env/dolphinscheduler_env.sh [hunk 2]
```diff
@@ -43,5 +43,6 @@ export PYTHON_HOME=${PYTHON_HOME:-/opt/soft/python}
 export HIVE_HOME=${HIVE_HOME:-/opt/soft/hive}
 export FLINK_HOME=${FLINK_HOME:-/opt/soft/flink}
 export DATAX_HOME=${DATAX_HOME:-/opt/soft/datax}
+export SEATUNNEL_HOME=${SEATUNNEL_HOME:-/opt/soft/seatunnel}
 
 export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$PYTHON_HOME/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_HOME/bin:$PATH
```
Comment (zhongjiajie) [inline script/env/dolphinscheduler_env.sh]: It is seems we missing adding it to $PATH
```suggestion
export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$PYTHON_HOME/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_HOME/bin:$SEATUNNEL_HOME/bin:$PATH
```

File: dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/DeployModeEnum.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.task.seatunnel;
+
+public enum DeployModeEnum {
+    cluster("cluster"),
+    client("client"),
+    local("client");
```
Comment (zhongjiajie) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/DeployModeEnum.java:23]: Is it a typo or it is truely should use `client`?

File: script/env/dolphinscheduler_env.sh (hunk 2 above)
Comment (zhuangchong) [reply script/env/dolphinscheduler_env.sh]: done.

File: dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/DeployModeEnum.java (hunk 3 above)
Comment (zhuangchong) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/DeployModeEnum.java:23]: `local` variables should use `client` mode

**Review by zhongjiajie — DISMISSED** [review]
LGTM, please merge latest change to solve the conflict

**Review by caishunfeng — DISMISSED** [review]
LGTM

File: docs/configs/docsdev.js [hunk 4]
```diff
@@ -141,6 +141,10 @@ export default {
                                 title: 'Switch',
                                 link: '/en-us/docs/dev/user_doc/guide/task/switch.html',
                             },
+                            {
+                                title: 'SeaTunnel',
+                                link: '/en-us/docs/dev/user_doc/guide/task/seatunnel.html',
+                            },
```
Comment (zhongjiajie) [inline docs/configs/docsdev.js:147]: we should add both En and Zh docs to docsdev,js

**Review by zhongjiajie — APPROVED** [review]
LGTM

**Comment by zhuangchong** [comment]
It has been merged into the 3.1.0 branch.

**Comment by RickyLoveFreedom** [comment]
can support Seatunnel new Seatunnel Engine ?

**Comment by zhuangchong** [comment]
> can support Seatunnel new Seatunnel Engine ?

Custom configuration can support.

**Pull request merged**